src/LazyLoadImage/LazyLoadImage.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useRef, useState } from "react";
+
+import type { LazyLoadImageProps } from "./LazyLoadImage.types";
+
+// TODO: add tests in browser once we are done with the migration to cypress
+
+/**
+ * Lazy load wrapper for HTML image brackets. Used for lazy loading images.
+ * @param LazyLoadImgProps The input object of the component
+ * @param LazyLoadImgProps.src string Source of the image
+ * @param LazyLoadImgProps.alt string Alternative text for the image
+ * @param LazyLoadImgProps.ImgProps object ImgProps Any other prop of the <img> component
+ */
+export default function LazyLoadImage({
+  src,
+  alt,
+  ImgProps
+}: LazyLoadImageProps) {
+  // state to track if the image is visible
+  const [isVisible, setIsVisible] = useState(false);
+
+  // reference to the image element
+  const imgRef = useRef(null);
+
+  // when the image is in view, set isVisible to true
+  useEffect(() => {
+    const observer = new IntersectionObserver(
+      entries => {
+        if (entries[0].isIntersecting) {
+          setIsVisible(true);
+          observer.disconnect();
+        }
+      },
+      { threshold: 0.1 }
+    );
+    if (imgRef.current) {
+      observer.observe(imgRef.current);
+    }
+    return () => observer.disconnect();
+  }, []);
+
+  // render the image
+  return (
+    <img
+      alt={alt || ""}
```

Comment:
This works fine, Do you think adding a loader animation something like this https://mui.com/material-ui/react-skeleton/#pulsate-example for the image @Sowbhagya-ipg would be nice?